### ISSUE TITLE
save_resume makes results worse

### ISSUE THREAD

**Issue by martinpopel** [VowpalWabbit/vowpal_wabbit#262]
When you want to train N subsequent models on N data sets, you must use --save_resume flag in the first N-1 trainings, but you SHOULD NOT use it in the last (N-th) training, if you want to get the same results as when training on all the N data sets concatenated. John Langford [confirmed that "this looks bugly"](https://groups.yahoo.com/neo/groups/vowpal_wabbit/conversations/topics/3329).

I attach an example with N=2.
Not using --save_resume makes the final test loss (0.223781) only a bit worse than the baseline (0.225275).
However, using --save_resume in both trainings makes the final test loss much worse (0.267824).

```
### prepare data train=set00,set01 test=set02
cd vowpal_wabbit/test/train-sets
split -dl 4000 rcv1_small.dat set

### train on concatenated training sets
cat set00 set01 | vw --loss_function=logistic -f final.model
cat set02 | vw --loss_function=logistic -i final.model -t
# average loss = 0.225275

### train separately, first with --save_resume, second without
cat set00 | vw --loss_function=logistic -f A.model --save_resume
cat set01 | vw --loss_function=logistic -i A.model -f final.model
cat set02 | vw --loss_function=logistic -i final.model -t
# average loss = 0.225275

### train separately, both models with --save_resume
cat set00 | vw --loss_function=logistic -f A.model --save_resume
cat set01 | vw --loss_function=logistic -i A.model --save_resume -f final.model
cat set02 | vw --loss_function=logistic -i final.model -t
# average loss = 0.267824

### train separately, without --save_resume
cat set00 | vw --loss_function=logistic -f A.model
cat set01 | vw --loss_function=logistic -i A.model -f final.model
cat set02 | vw --loss_function=logistic -i final.model -t
# average loss = 0.223781
```
